crates/orchestrator/src/jobs/mod.rs

```diff
@@ -173,6 +173,18 @@ pub async fn verify_job(id: Uuid) -> Result<()> {
     Ok(())
 }
 
+/// Terminates the job and updates the status of the job in the DB.
+pub async fn terminate_job(id: Uuid) -> Result<()> {
```

Comment:
```suggestion
pub async fn handle_job_failure(id: Uuid) -> Result<()> {
```

Reply:
terminate sounds like stopping a job in the middle, wdyt?

Reply:
No issues, changing to handle_job_failure